File: pt2pt/Go/startup.go
package main

import (
	"fmt"
	"go-mpi/MPI"
	"os"
	"time"
)

func main() {

	var myid int
	var t_start time.Time
	var t float64
	t_start = time.Now()
	MPI.Init(&os.Args)
	t = float64(time.Now().Sub(t_start).Nanoseconds()) / 1e3
	myid, _ = MPI.Comm_rank(MPI.COMM_WORLD)
	if myid == 0 {
		fmt.Fprintf(os.Stdout, "%*.*f\n", FIELD_WIDTH, FLOAT_PRECISION, t)
	}
	MPI.Finalize()
	os.Exit(0)
}

File: pt2pt/C/startup.c
#ifndef BENCHMARK_HEADER
	#include "benchmark.h"
#endif

int
main (int argc, char *argv[])
{
	int myid;
	struct timeval t_start;
    struct timeval t_end;
    double t = 0.0;
    gettimeofday(&t_start, NULL);
    MPI_Init(&argc, &argv);
    gettimeofday(&t_end, NULL);
    t = 1e6 * (t_end.tv_sec - t_start.tv_sec) + t_end.tv_usec- t_start.tv_usec;
    MPI_Comm_rank(MPI_COMM_WORLD, &myid);
    if(myid == 0)
    	fprintf(stdout, "%*.*f\n", FIELD_WIDTH, FLOAT_PRECISION, t);
    fflush(stdout);
    MPI_Finalize();
    exit(EXIT_SUCCESS);
}